git/git.go

```diff
@@ -203,6 +203,18 @@ func ReadBranchConfig(branch string) (cfg BranchConfig) {
 	return
 }
 
+func DeleteLocalBranch(branch string) error {
+	configCmd := GitCommand("branch", "-d", branch)
```

Comment:
⚠️ We most likely want to `git branch -D` here instead of `git branch -d`.

`git branch -d` will fail if the branch to be deleted is not yet detected to be "merged" in its upstream branch. Even though we just merged this branch in the remote, the local repo is not aware this operation just yet, to this command will always fail.

Furthermore, if squashed or rebased merge options are used, the branch will never qualify as "merged" in git terms. Therefore it's better to use `-D` so we can be sure that the branch will be deleted regardless.

Reply:
hm; I put in some debugging and see that local branch deletion does succeed:

![image](https://user-images.githubusercontent.com/98482/82690055-25305600-9c21-11ea-951a-85545dacea6b.png)

It still sounds wise to use `-D` but I'd like to understand why it doesn't always fail as you suggest.

Reply:
I've been running the command A LOT and it has yet to fail. When you try to do this locally I get this warning but the command is still successful:

```
warning: deleting branch 'branch-10471' that has been merged to
         'refs/remotes/origin/branch-10471', but not yet merged to HEAD.
Deleted branch branch-10471 (was 477cd32).
```

The odd part is, when I look at the output from the git command run by `gh` it doesn't show this error. I'm assuming this is has something to do with the command not capturing stderr, but I tried to verify that without success.

For now I've switched it to `-D` because that won't seem to have any effect either way.